client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/HotrodCache.java

```diff
@@ -162,44 +162,45 @@ public void addReconnectListener(final ReconnectListener<HotrodCache<K, V>> list
     @Override
     public Future<V> put(final K key, final V value) {
 
-        final Promise<V> result = Promise.promise();
         if (cache == null) {
-            result.fail(new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE, "no connection to data grid"));
+
+            return noConnectionFailure();
+
         } else {
-            vertx.executeBlocking(outcome -> {
-                try {
-                    final V replacedValue = cache.put(key, value);
-                    outcome.complete(replacedValue);
-                } catch (final Throwable e) {
-                    outcome.fail(e);
-                }
-            }, result);
+
+            return MoreFutures.create(() -> {
```

Comment:
I find the naming of `MoreFutures.create()` somewhat unintuitive (are more/multiple Futures created here?).
Looking at the Javadoc of the class and the method, I think something like `FuturesHelper.mapCompletionStageResult()` would make it more obvious what is being done here. WDYT?

Reply:
I followed this pattern: https://guava.dev/releases/23.0/api/docs/com/google/common/base/MoreObjects.html

But I am open to making changes to the name.

Reply:
I think with `MoreObjects` that pattern works, because it complements the `Objects` helper class. Since there is no `Futures` helper class for Vert.x or Java futures, this pattern doesn't work so well here, IMHO.

Reply:
Then why not simply call the class `Futures`?

Reply:
Because we already have a class called `Futures` in our classpath: `com.google.common.util.concurrent.Futures`.

Reply:
Assuming you still prefer `Futures`, I updated the pull request to that name. If you re-consider, I can still revert back to `MoreFutures`.